rust/lance/src/io/exec/pushdown_scan.rs

```diff
@@ -294,7 +294,12 @@ impl FragmentScanner {
         let stream = futures::stream::iter(simplified_predicates.into_iter().enumerate()).map(
             move |(batch_id, predicate)| {
                 let scanner_ref = scanner.clone();
-                async move { scanner_ref.read_batch(batch_id, predicate).await }
+                tokio::task::spawn(async move { scanner_ref.read_batch(batch_id, predicate).await })
```

Comment:
Would `buffered` / `buffer_unordered` not work here?

Reply:
We do call it below, it's just not sufficient. Each of the `read_batch()` tasks has some CPU-bound work, which can block IO of other tasks if they are in the same thread. By spawning, we distribute the tasks amongst threads, ensuring we get concurrency.